### DISCUSSION TITLE
Torrent Download Speed Stalling Out / Slowing

### DISCUSSION THREAD

**Discussion by Chewyyy** [tteck/Proxmox#3593] [Help Needed] [answered]
Hey all, I have a setup right now that is a strict Docker media server for the -arr's and using qbitt.

I am upgrading to a newer machine, and using proxmox and have been running into consistent issues related to torrent downloads...  I've done proxmox containers for the -arr programs and qbitt (and deluge), and when I select a download (even manually with a lot of seeders) the speed ramps up to 40-50 Mb and then drops down to the KiB range and/or stalls out completely.

I've tried removing VPNs, removing any tunneling, and no luck.

I've even used portainer inside a container and made docker containers (this was for troubleshooting) and they do the same thing as before

I know it isn't my ISP because if I do the same download on my old machine, which is an ubuntu docker setup, and the speeds stay up and the downloads finish.

Both machines are downloading torrents to the same NAS.  The NAS is mounted to the working machine, and the NAS is setup and mounted to the bittorrent LXC (I followed this video for the setup: https://youtu.be/DMPetY4mX-c?si=v3CYgDjq1Sprspvl)

I had a previous rendition of my setup (that I erased and started anew due to frustrations) where I added my NAS through SMB under the host, and created mount points / permissions, and I had the same issue.

ANY assistance is greatly appreciated.  I am at an utter loss.

**Comment by Chewyyy** [comment]
Just as an added troubleshooting point, I did a dd test from the host and got 70.6MB/s read and 83 MB/s write, and 72MB/81MB on the qBittorrent LXC container

**Comment by Mellowlynx** [comment]
Speeding up at first and then dropping down, sounds like a storage/Io issue.

**Reply by Chewyyy** [reply]
Any troubleshooting methods I'm missing?

I'm using the same NAS with my other system, and the downloads are fine

**Reply by Mellowlynx** [reply]
You could try to check in Proxmox UI, just go to the node where the LXC is on and check it's CPU/IO Delay on the Summery page.
It should be in the summary and also in the CPU usage graph.
Start a download and check what is happening on that page….
If you see no change in the IO part, then the drive is probably fine, if you see change, my next step would be to check the SMART status of the drive.

**Reply by Chewyyy** [reply]
Both under a percentage point, speeds started at 2MB/s, tapered down to 4B/s then stalled. (This test was done without a VPN)

**Reply by Mellowlynx** [reply]
Do you directly download to your NAS or save the temp local?
You could also try a normal, bigger download, maybe? wget a iso file of something
Just to see how that speed does when it saves local.

But without seeing it myself, it's a bit difficult to troubleshoot blind.

**Reply by Chewyyy** [reply]
So I'm downloading directly to my NAS, which I think I just figured out might be the problem (even though it works on my other setup).

I just remade my qbitt LXC container, and allocated more SSD space to it, and the download flew by and finished insanely quick.  Now waiting to see if radarr appropriately moves it to the mounted network drive.

Is this the preferred method? Or should I try and figure out why torrent clients cant download direct to my NAS?

**Reply by Mellowlynx** [reply, marked as answer]
My -arr's setup is on a Windows 11-tiny installation. (server 2019 before)
I like this more because all the -arr's in a separate LXC requires more setup/maintenance and recourses than have it all on just one Windows VM.
I also use that VM to run the downloader, backup my mails to, and some other smaller stuff I need from time to time.
Qbitt moves it to my NAS when done, and then radarr/sonarr/etc. moves is to the right spot on the NAS.
(Sidenote: All internet traffic goes through a VPN, but overall a way easier setup then trying to add a VPN to a LXC.)

In my experience, there is no big downside to have it download directly to the NAS or not.
I did both in the past, but ended up with local first to give my network a bit more bandwidth to and from that node.
If both the NAS and PVE node are on the same switch, it might save you space on the LXC/VM if you download directly to the NAS.
This might be the most considerable difference in most cases, I think.

**Reply by Chewyyy** [reply]
Thanks for the information!

Just for any future researchers:

Essentially what I had to do was: 
1. Create a downloads folder on local storage (NVME), then mount those drives to radarr/sonarr/qbitt.
2. Give permissions to the containers to access the local storage location (for those not caring about security 'chmod 777 -R /your/path/to/downloads
3. Setup Remote Path Mappings (Guide: https://trash-guides.info/Radarr/Radarr-remote-path-mapping/) mapping your downloads folder to radarr/sonarr (this only needs to be done if they are all not on the same system ie: separate lxc containers)
4. Add new downloads location to the mounted drive in qbitt (or torrent of choice)

That fixed it for me.  My downloads go through, store on local drive in created downloads folder, then transfer to my NAS for media storage which is accessible by Plex.

@Mellowlynx Thanks again for pointing me in the right direction!!!
